### ISSUE TITLE
Progress bar does NOT disappear

### ISSUE THREAD

**Issue by ranchop** [tqdm/tqdm#352]
Consider the following code in Jupyter Notebook.
`for i in tqdm(range(100), leave=False): pass`
The progress bas is supposed to disappear after execution, but it does NOT. I noticed this happens whenever the loop is executed very quickly. 

I have only tested this on Jupyter Notebook using both Safari and Chrome and Python 3.

**Comment by richlewis42** [comment]
Similar with the html progress bars: for very short loops:

```python
from tqdm import tqdm_notebook
from time import sleep

with tqdm_notebook(range(100), desc='loop1', leave=True) as bar:
    for i in bar:
        with tqdm_notebook(range(5), desc='loop2', leave=False) as jar:
            for j in jar:
                sleep(0.01)
```
Leaves all the progress bars.

Tested with Safari and Firefox on Mac OSX, python 3.5, tqdm 4.11.2.

**Comment by peporiz0r** [comment]
I noticed something similar, with nested loops in terminal.

```
for i in range(10):
    for j in range(859):
        do something
```

Everthing the inner loop has finished, the status bar for the new outer loop is not updated. Instead a new bar gets drawn.

![screenshot 2017-03-21 14 37 45](https://cloud.githubusercontent.com/assets/18383180/24149842/34e441c4-0e44-11e7-8d6c-30c427a4cb2e.png)

**Comment by michaelosthege** [comment]
Does someone know the version number of the latest build that did work? (nested bars in Jupyter notebooks)

**Comment by lrq3000** [comment]
It was with Jupyter 4 and ipywidgets 4, both the first releases in this
major branch, I don't know why it does not work anymore, something in the
ipywidgets api probably changed. I'll check it out later.

2017-03-29 3:26 GMT+02:00 michaelosthege <notifications@github.com>:

**Comment by nunocalaim** [comment]
Same issue here! I just installed tqdm, and I'm using the same work environment as @richlewis42

**Comment by dvm-shlee** [comment]
Same here, really hope this issue could be solved for next releases.

**Comment by lzkelley** [comment]
@casperdcl @lrq3000 if the fix by @dvm-shlee works, can we get it pulled in?

**Comment by casperdcl** [comment]
closed by c0a402a138629eff37efe2002f05cef2793efd1c
